tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/AuthorityTests.cs

```diff
@@ -88,6 +88,33 @@ public async Task FailedAuthorityValidationTestAsync()
             Assert.AreEqual("invalid_instance", exception.ErrorCode);
         }
 
+        [TestMethod]
+        public async Task AuthorityValidationTestWithFalseValidateAuthorityAsync()
+        {
+            LabResponse labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            LabUser user = labResponse.User;
+
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+                .Create(labResponse.App.AppId)
+                .WithAuthority("https://bogus.microsoft.com/common", false)
+                .WithTestLogging()
+                .Build();
+
+            Trace.WriteLine("Acquire a token using a not so common authority alias");
+
+            HttpRequestException exception = await AssertException.TaskThrowsAsync<HttpRequestException>(() =>
+                 pca.AcquireTokenByUsernamePassword(
+                    s_scopes,
+                     user.Upn,
+                     new NetworkCredential("", user.GetOrFetchPassword()).SecurePassword)
+                     .ExecuteAsync())
+                .ConfigureAwait(false);
+
+
+            Assert.IsTrue(((System.Net.WebException) exception.InnerException).Status == System.Net.WebExceptionStatus.NameResolutionFailure);
```

Comment:
we don't get a better exception here?

Reply:
No. What happens here is first there will be network discovery and if that fails it will check the validateAuthority flag. If the flag is false it will go directly to the uri. So, the authority here is a fake uri and that is what the exception is. In a normal scenario there will be no exception and a token will be generated.

Reply:
we should catch that exception and through an MsalServiceException (as this is a service issue, but maybe a clientException... @bgavrilMS what do you think?). See my [this comment](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1817#issuecomment-628342946) first though :)

Reply:
`MsalServiceException` is meant to describe AAD service errors. We have a contract with AAD and we know how to interpret these errors (i.e. read the JSON, with error_code, message etc.)

In this case, the error isn't coming from AAD. The request never reaches AAD. It goes into the great unknown that is the Internet and it dies a 404 Not Found death. MSAL is doing the right thing to not try to interpret this error - it just lets it bubble up to the developer. After all, it's the developer who used "bogus.microsoft.com" as authority, what do they expect? :D

So I think this behavior is ok. The test shows that instance discovery isn't blocking request.

As for disabling OIDC, you are right Jenny, I made changes. But I think my changes are much more extensive, so I will rebase them over Neha's change rather than the other way around.